core/guests.core.js

```diff
@@ -84,26 +88,59 @@ async function saveNewGuestWithEventId(userId, eventId, guestInfo) {
     },
   };
 
+  if (!emailVerified) {
+    const verifyEmail = new VerifyGuestEmail({
+      to: `${newGuest.email}`,
+      html: `Xin chào ${newGuest.email} <br/>
+      Bạn đã đăng ký tham gia sự kiện ${event.name}. <br/>
+      Chúng tôi cần bạn xác nhận lại email. Vui lòng ấn vào đường dẫn sau để xác nhận: <br/>
+      <a href="#">link</a><br/>
+      Chúng tôi sẽ gửi thông tin vé sớm nhất cho bạn`,
+    });
+
+    mailgun.send(verifyEmail);
+  }
+
   const savedGuest = await guestODM.save(newGuest);
   return {
     savedGuest,
   };
 }
 
+async function sendTicketMail(guest, event, ticketCode) {
+  const dataImage = await QRCode.generateBase64Buffer(ticketCode);
+
+  const ticketMail = new TicketEmail({
+    to: `${guest.email}`,
+    html: `Xin chào ${guest.email}, lại là Easy Event đây !<br/>
+    Sau đây là vé dành cho sự kiện ${event.name} <br/>
+    <ul>
+      <li>Email: ${guest.email}</li>
+      <li>Gender: ${guest.info.gender}</li>
+    </ul>
+    <br/>
+    Bạn hãy vui lòng sử dụng mã QR Code trong file đính kèm để check in tại sự kiện nhé !`,
+    attachment: new mailgun.default.Attachment({ data: dataImage, filename: 'ticket.png' }),
+  });
+  await mailgun.send(ticketMail);
+}
+
 /**
  *
  * @param {String} guestId
  * @param {String} action
  */
 async function updateGuest(guestId, action) {
-  const guest = guestODM.findById(guestId);
+  const guest = await guestODM.findById(guestId);
   if (!guest) {
     throw new GuestNotFoundError();
   }
+  const event = await eventODM.findById(guest.get('event'));
   let updates = {};
+  let ticketCode = null;
   switch (action) {
```

Comment:
`switch (action)` should be in controller. Split `core.updateGuest` to many functions for each action.

Reply:
Solved: https://github.com/EasyEventV2/ee2_api/pull/9/commits/687280975dfd81b4b38ece65fe0eb712f43dc07c